pkgs/applications/misc/sweethome3d/linux.nix

```
@@ -0,0 +1,132 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+  fetchurl,
+  makeWrapper,
+  makeDesktopItem,
+  jdk,
+  ant,
+  stripJavaArchivesHook,
+  gtk3,
+  gsettings-desktop-schemas,
+  p7zip,
+  autoPatchelfHook,
+  libXxf86vm,
+  libGL,
+
+  pname,
+  version,
+  src,
+  meta,
+}:
+
+let
+
+  # TODO: Should we move this to `lib`? Seems like its would be useful in many cases.
+  extensionOf =
+    filePath: lib.concatStringsSep "." (lib.tail (lib.splitString "." (builtins.baseNameOf filePath)));
+
+  installIcons =
+    iconName: icons:
+    lib.concatStringsSep "\n" (
+      lib.mapAttrsToList (size: iconFile: ''
+        mkdir -p "$out/share/icons/hicolor/${size}/apps"
+        ln -s -T "${iconFile}" "$out/share/icons/hicolor/${size}/apps/${iconName}.${extensionOf iconFile}"
+      '') icons
+    );
+
+  icons = {
+    "32x32" = fetchurl {
+      url = "http://sweethome3d.cvs.sourceforge.net/viewvc/sweethome3d/SweetHome3D/deploy/SweetHome3DIcon32x32.png";
+      sha256 = "1r2fhfg27mx00nfv0qj66rhf719s2g1vhdis7bdc9mqk9x0mb0ir";
+    };
+    "48x48" = fetchurl {
+      url = "http://sweethome3d.cvs.sourceforge.net/viewvc/sweethome3d/SweetHome3D/deploy/SweetHome3DIcon48x48.png";
+      sha256 = "1ap6d75dyqqvx21wddvn8vw2apq3v803vmbxdriwd0dw9rq3zn4g";
+    };
+  };
+
+in
+stdenv.mkDerivation (finalAttrs: {
+  inherit pname version src;
+
+  sweethome3dItem = makeDesktopItem {
+    desktopName = "Sweet Home 3D";
+    name = finalAttrs.pname;
+    icon = finalAttrs.pname;
```

```suggestion
    name = "sweethome3d";
    icon = "sweethome3d";
```
any change to this is wrong